cvat/apps/engine/models.py

```diff
@@ -514,7 +514,8 @@ def has_parent_label(self):
 
     class Meta:
         default_permissions = ()
-        unique_together = ('task', 'name', 'parent')
+        unique_together = (('task', 'name', 'parent'),
+                           ('project', 'name', 'parent'))
```

Comment:
During experimenting, I've noticed that this constraint doesn't work for some reason. I've tried the following variants:

```
unique_together = (('task', 'name', 'parent'), ('project', 'name', 'parent'))
unique_together = (('task_id', 'name', 'parent_id'), ('project_id', 'name', 'parent_id'))
unique_together = ('project', 'task', 'name', 'parent')
unique_together = ('project_id', 'task_id', 'name', 'parent_id')
unique_together = ('name', 'parent')
unique_together = ('name', 'parent_id')
constraints = [models.UniqueConstraint(fields=...all the same..., name=...)]

unique_together = ('name', ) # only this worked
```
Only the simplest one worked in my case. If `validate_labels()` checks are removed in `Task/Project- WriteSerializer`, users are able to create several equal labels. This problem doesn't appear in other cases where `unique_together` used. SO tells it can be because of the DB used, but both with MySQL and Postgres the problem appears.

The test case (remember to remove `validate_labels`):
POST /api/projects

```
{
  "name": "test_duplicate_labels",
  "labels": [{"name": "cat"}, {"name": "cat"}, {"name": "cat"}]
}
```
-> OK, 3 same labels in a project.

Reply:
I guess it can happen because we have 3 ForeignKey which can be null. To solve the problem with UniqueConstraint, all permutations should be specified (but it's too hard, may be we can use `UniqueTogetherValidator`).
```
constraints = [
    models.UniqueConstraint(
        name='project_name_parent_unique',
        fields=('project', 'name', 'parent'),
        condition=Q(task__isnull=True)
    ),
    ...
    models.UniqueConstraint(
        name='name_unique',
        fields=('name',),
        condition=Q(project__isnull=True) & Q(task__isnull=True) & Q(parent__isnull=True)
    )
]
```

Reply:
Is there any mentioning in the documentation about problems with `null`? 
For instance, this tuple should work normally even with nulls:

```
unique_together = ('project', 'task', 'name', 'parent')
```

because it makes unique hashes for each label.

Reply:
In SQL null is not equal to anything - not even to another null

Reply:
Okay, probably, this resolves the question then.